### DISCUSSION TITLE
Enter Auth Codes not working periodically

### DISCUSSION THREAD

**Discussion by bren924** [ente-io/ente#3434] [Q&A] [answered]
I've recently run into an issue in Ente Auth where the code displayed is not working in the target site. When I wait for the codes to roll over, I also notice that the code that should be next is not the one that becomes active, but the newly displayed code is valid. 

It seems the codes are off slightly when opening the app after it has been closed for a period of time.

This is on iPhone 13 iOS 18, latest version of the Ente Auth app.

**Comment by LaczPro19** [comment]
Same here. I want to suggest something to refresh the codes because it looks like the codes are getting way ahead of what they should be. Comparing the codes from my phone to the desktop, it seems the codes on the PC get way earlier than those on my phone (which is correct).

Update: Nope. The app has nothing to do with that. **Always check the time and date settings are up to date, and if not, refresh them**.

**Comment by gengurke** [comment]
Same here on Android. Once the first code rollover happens the codes are fine.

I imported all my codes from Google authenticator originally.

Ente Auth 4.0.0
Android 13

**Comment by j-onty** [comment]
same problem here

**Comment by sudokit** [comment]
Yeah same problem here especially with the web version. Kinda scary...
EDIT: turns out my system clock was just a bit off.

**Comment by brog-io** [comment, marked as answer]
This issue is likely because your system clock is out of sync. Please check your device settings and ensure your date and time are set correctly, preferably with automatic time synchronization enabled.
<img width="843" height="118" alt="image" src="https://github.com/user-attachments/assets/61625326-c111-42b5-8484-cf005e328789" />

**Reply by sudokit** [reply]
Mm yeah that was the issue.
